### PR TITLE
Fix the bug where colon is replaced by underscore.

### PR DESCRIPTION
**Pull request by XianXing**
More details can be find in issue #135.

Unit test also added to cover the bug.

### CODE REVIEW

**Comment by XianXing** [comment]
@timyitong

File: photon-ml/src/test/scala/com/linkedin/photon/ml/cli/game/training/ParamsTest.scala [hunk 1]
```diff
@@ -334,44 +345,48 @@ object ParamsTest {
...
-    args ++ requiredArgs()
+  def setOneMoreArg(argName: String, argValue: String): Map[String, String] = {
+    val name = CommonTestUtils.fromOptionNameToArg(argName)
+    val value = argValue
+    requiredArgs().updated(name, value)
+  }
+
+  implicit def mapToArray(args: Map[String, String]): Array[String] = {
+    args.toArray.flatMap { case (name, value) => Seq(name, value) }
   }
```
Comment (XianXing) [inline photon-ml/src/test/scala/com/linkedin/photon/ml/cli/game/training/ParamsTest.scala]: I usually prefer not to use implicit function in Scala for readability and performance reasons, but the implicit conversion comes pretty handy here for unit tests.

**Comment by ashelkovnykov** [comment]
LGTM

File: photon-ml/src/test/scala/com/linkedin/photon/ml/cli/game/training/ParamsTest.scala (hunk 1 above)
Comment (joshvfleming) [inline photon-ml/src/test/scala/com/linkedin/photon/ml/cli/game/training/ParamsTest.scala]: Yeah. I agree that this kind of thing is okay for making tests more concise, but could lead to readability and comprehensibility problems in the production code.

File: photon-ml/src/main/scala/com/linkedin/photon/ml/cli/game/training/Params.scala [hunk 2]
```diff
@@ -216,7 +216,7 @@ object Params {
       opt[String]("output-dir")
         .required()
         .text(s"Output directory for logs and learned models.")
-        .foreach(x => params.outputDir = x.replace(',', '_').replace(':', '_'))
+        .foreach(x => params.outputDir = x.replace(',', '_'))
```
Comment (joshvfleming) [inline photon-ml/src/main/scala/com/linkedin/photon/ml/cli/game/training/Params.scala]: We could make this a little more robust by having it parse URIs first, and then do the replacement -- if the intention is to support URIs as an input here (and if it's important to do the colon replacement).

I don't think we need to do this now though.

**Comment by timyitong** [comment]
@joshvfleming: Thanks so much for fixing this so quickly!

**Pull request merged**